### PR TITLE
Allow otpUrl to generate TOTP/HOTP tokens

### PR DESCRIPTION
**Pull request by Mikescops**
Reported by team admin through user support, some credentials with OTP where not working.

This is due to a new OTP URL format that was not taken into account in the CLI code.

### CODE REVIEW

File: src/command-handlers/passwords.ts [hunk 1]
```diff
@@ -28,7 +27,7 @@ export const runPassword = async (
     }
 
     if (field === 'otp') {
-        foundCredentials = foundCredentials.filter((credential) => credential.otpSecret);
+        foundCredentials = foundCredentials.filter((credential) => credential.otpSecret || credential.otpUrl);
 
         if (foundCredentials.length === 0) {
```
Comment (showi) [inline src/command-handlers/passwords.ts:32]: Why you don't exit before the if, the rest of the function will have no purpose with no result or ?

File: src/modules/crypto/otpauth.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            return { token: authenticator.generate(otpauth.secret), remainingTime: authenticator.timeRemaining() };
+        case 'hotp':
+            hotp.options = { algorithm: otpauth.algorithm, digits: otpauth.digits };
+            return { token: hotp.generate(otpauth.secret, otpauth.counter), remainingTime: null };
+        default:
+            throw new Error('Invalid OTP type');
+    }
+};
+
+export const generateOtpFromSecret = (secret: string): { token: string; remainingTime: number | null } => {
```
Comment (showi) [inline src/modules/crypto/otpauth.ts:71]: You could have type the returns.

**Review by showi — APPROVED** [review]

File: src/command-handlers/passwords.ts (hunk 1 above)
Comment (Mikescops) [reply src/command-handlers/passwords.ts:32]: 🤓

File: src/modules/crypto/otpauth.ts (hunk 2 above)
Comment (Mikescops) [reply src/modules/crypto/otpauth.ts]: thanks done

**Pull request merged**